hello, this is an answer for Notebook 1

steps, 
1. read the data
2. visualize the data
3. clean the data
4. merge the data
5.answer the questions


In [6]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

### read (Patients data)

In [11]:
df_cp = pd.read_csv('data/data_clinical_patient.txt', delimiter='	' )
print("Data shape: "+str(df_cp.shape))
df_cp.head(5)

Data shape: (415, 37)


#Patient Identifier  Subtype  \
0  #Identifier to uniquely specify a patient.  Subtype   
1                                     #STRING   STRING   
2                                          #1        1   
3                                  PATIENT_ID  SUBTYPE   
4                                TCGA-2F-A9KO     BLCA   

                TCGA PanCanAtlas Cancer Type Acronym  \
0  Text field to hold cancer type acronym used by...   
1                                             STRING   
2                                                  1   
3                                CANCER_TYPE_ACRONYM   
4                                               BLCA   

                          Other Patient ID  \
0  Legacy DMP patient identifier (DMPnnnn)   
1                                   STRING   
2                                        1   
3                         OTHER_PATIENT_ID   
4     DD3A6357-9087-44B2-9956-F981E0DE6F1C   

                                       Diagnosis Age     Sex  \
0  Age at which a condition or disease was first ...     Sex   
1                                             NUMBER  STRING   
2                                                  1       1   
3                                                AGE     SEX   
4                                                 63    Male   

  Neoplasm Disease Stage American Joint Committee on Cancer Code  \
0  The extent of a cancer, especially whether the...               
1                                             STRING               
2                                                  1               
3                        AJCC_PATHOLOGIC_TUMOR_STAGE               
4                                           STAGE IV               

  American Joint Committee on Cancer Publication Version Type  \
0  The version or edition of the American Joint C...            
1                                             STRING            
2                                                  1            
3                               AJCC_STAGING_EDITION            
4                                                6TH            

  Last Communication Contact from Initial Pathologic Diagnosis Date  \
0  Time interval from the date of last followup t...                  
1                                             NUMBER                  
2                                                  1                  
3                                 DAYS_LAST_FOLLOWUP                  
4                                                678                  

        Birth from Initial Pathologic Diagnosis Date  ...  \
0  Time interval from a person's date of birth to...  ...   
1                                             NUMBER  ...   
2                                                  1  ...   
3                                      DAYS_TO_BIRTH  ...   
4                                             -23323  ...   

                  Patient Weight               In PanCan Pathway Analysis  \
0  Weight measured in kilograms.  Patient Part of PanCan Pathway Analysis   
1                         NUMBER                                   STRING   
2                              1                                        1   
3                         WEIGHT                 IN_PANCANPATHWAYS_FREEZE   
4                             65                                      Yes   

            Overall Survival Status  \
0  Overall patient survival status.   
1                            STRING   
2                                 9   
3                         OS_STATUS   
4                        1:DECEASED   

                           Overall Survival (Months)  \
0  Overall survival in months since initial diago...   
1                                             NUMBER   
2                                                  1   
3                                          OS_MONTHS   
4                                         24.1312424   

                    Disease-specific Survival sta

notice, the 4 four rows are duplicated, therfore we have to delete those rows

In [12]:
df_cp.columns = df_cp.iloc[3] # fix the col names to be the 3rd col.
df_cp = df_cp.iloc[4:] # for the rows
print("Data shape: "+str(df_cp.shape))
df_cp.head(5)

Data shape: (411, 37)


3    PATIENT_ID SUBTYPE CANCER_TYPE_ACRONYM  \
4  TCGA-2F-A9KO    BLCA                BLCA   
5  TCGA-2F-A9KP    BLCA                BLCA   
6  TCGA-2F-A9KQ    BLCA                BLCA   
7  TCGA-2F-A9KR    BLCA                BLCA   
8  TCGA-2F-A9KT    BLCA                BLCA   

3                      OTHER_PATIENT_ID AGE     SEX  \
4  DD3A6357-9087-44B2-9956-F981E0DE6F1C  63    Male   
5  65D1EAEC-28DB-4A41-A4DB-1A710FCB24AD  66    Male   
6  25EAF3F9-C364-423C-AAAE-925E7B393AFC  69    Male   
7  F6D916B0-8E4C-49CB-A0D4-883908F3284F  59  Female   
8  BC6C516B-591E-4950-B6B0-DECAFA666F4E  83    Male   

3 AJCC_PATHOLOGIC_TUMOR_STAGE AJCC_STAGING_EDITION DAYS_LAST_FOLLOWUP  \
4                    STAGE IV                  6TH                678   
5                    STAGE IV                  7TH                NaN   
6                   STAGE III                  6TH               2886   
7                   STAGE III                  5TH               3148   
8                    STAGE II                  6TH               3314   

3 DAYS_TO_BIRTH  ... WEIGHT IN_PANCANPATHWAYS_FREEZE   OS_STATUS    OS_MONTHS  \
4        -23323  ...     65                      Yes  1:DECEASED   24.1312424   
5        -24428  ...    130                      Yes  1:DECEASED  11.96699214   
6        -25259  ...     72                      Yes    0:LIVING  94.88115199   
7        -21848  ...     80                      Yes  1:DECEASED  104.6454285   
8        -30520  ...     80                      Yes    0:LIVING  108.9522307   

3                  DSS_STATUS   DSS_MONTHS             DFS_STATUS  \
4           1:DEAD WITH TUMOR   24.1312424                    NaN   
5           1:DEAD WITH TUMOR  11.96699214                    NaN   
6  0:ALIVE OR DEAD TUMOR FREE  94.88115199          0:DiseaseFree   
7           1:DEAD WITH TUMOR  104.6454285  1:Recurred/Progressed   
8  0:ALIVE OR DEAD TUMOR FREE  108.9522307          0:DiseaseFree   

3   DFS_MONTHS     PFS_STATUS   PFS_MONTHS  
4          NaN  1:PROGRESSION  21.40250518  
5          NaN  1:PROGRESSION  11.27658875  
6  94.88115199     0:CENSORED  94.88115199  
7  101.4564224  1:PROGRESSION  101.4564224  
8  108.9522307     0:CENSORED  108.9522307  

[5 rows x 37 columns]

### read (MRNA data)

In [44]:
df_mrna = pd.read_csv('data/TCGA_BLCA_RSEM_TPM.txt', delimiter = "	")
# df_mrna = pd.read_csv('data/data_mrna_seq_v2_rsem.txt', delimiter = "	")
print("Data shape: "+str(df_mrna.shape))
df_mrna.head(30)

Data shape: (20531, 429)


gene_symbol  entrez_gene_id  TCGA-2F-A9KO-01A-11R-A38B-07...3  \
0            ?       100130426                          0.000000   
1            ?       100133144                          1.613410   
2            ?       100134869                          2.179003   
3            ?           10357                         21.731101   
4            ?           10431                         57.752456   
5            ?          136542                          0.000000   
6            ?          155060                         12.348807   
7            ?           26823                          0.127691   
8            ?          280660                          0.000000   
9            ?          317712                          0.000000   
10           ?          340602                          0.518427   
11           ?          388795                          0.050426   
12           ?          390284                          2.057288   
13           ?          391343                          0.000000   
14           ?          391714                          0.000000   
15           ?          404770                          0.000000   
16           ?          441362                          0.000000   
17           ?          442388                          0.000000   
18           ?          553137                          0.271702   
19           ?           57714                         53.795822   
20           ?          645851                          2.745584   
21           ?          652919                          0.571500   
22           ?          653553                        554.997526   
23           ?          728045                          0.000000   
24           ?          728603                          0.000000   
25           ?          728788                          0.000000   
26           ?          729884                          0.875312   
27           ?            8225                         53.943808   
28           ?           90288                          0.048187   
29        A1BG               1                          1.407315   

    TCGA-2F-A9KP-01A-11R-A38B-07...6  TCGA-2F-A9KQ-01A-11R-A38B-07...9  \
0                           0.000000                          0.000000   
1                           1.349402                          1.124814   
2                           0.765463                          0.662142   
3                          25.957999                         19.792975   
4                          76.897486                         83.883269   
5                           0.000000                          0.000000   
6                           6.053803                          7.117069   
7                           0.194915                          0.000000   
8                           0.000000                          0.000000   
9                           0.000000                          0.000000   
10                          0.000000                          0.000000   
11                          0.104437                          0.010698   
12                          1.144199                          1.260638   
13                          0.000000                          0.000000   
14                          0.029198                          0.000000   
15                          0.000000                          0.000000   
16                          0.000000                          0.000000   
17                          0.000000                          0.000000   
18                          0.031511                          0.000000   
19                          4.649261                          5.761930   
20                         11.030052                          1.517669   
21                          0.000000                          0.000000   
22                        394.097860                        497.641806   
23                          0.000000                          0.000000   
24                          0.000000        

two things to notice,
1. there is the mismatch between the # of Patients between the two above data frames.
2. gene_symbol has lots of empty fields.

#### formatting mrna data

In [45]:
df_mrna['gene_symbol'] = df_mrna['gene_symbol'].replace('?', np.nan)
df_mrna.gene_symbol.fillna( df_mrna.entrez_gene_id, inplace=True)
del df_mrna['entrez_gene_id']
geneIDs = np.array(df_mrna['gene_symbol'])
print("Data shape: "+str(df_mrna.shape))
df_mrna.head(4)

Data shape: (20531, 428)


gene_symbol  TCGA-2F-A9KO-01A-11R-A38B-07...3  \
0   100130426                          0.000000   
1   100133144                          1.613410   
2   100134869                          2.179003   
3       10357                         21.731101   

   TCGA-2F-A9KP-01A-11R-A38B-07...6  TCGA-2F-A9KQ-01A-11R-A38B-07...9  \
0                          0.000000                          0.000000   
1                          1.349402                          1.124814   
2                          0.765463                          0.662142   
3                         25.957999                         19.792975   

   TCGA-2F-A9KR-01A-11R-A38B-07...12  TCGA-2F-A9KT-01A-11R-A38B-07...15  \
0                           0.000000                           0.000000   
1                           1.231898                           1.145208   
2                           2.484428                           1.050426   
3                          26.609935                          26.206662   

   TCGA-2F-A9KW-01A-11R-A38B-07...18  TCGA-4Z-AA7M-01A-11R-A39I-07...21  \
0                           0.000000                           0.000000   
1                           0.601180                           2.053420   
2                           0.141887                           2.992586   
3                          33.845656                          26.031972   

   TCGA-4Z-AA7N-01A-11R-A39I-07...24  TCGA-4Z-AA7O-01A-31R-A39I-07...27  ...  \
0                           0.000000                           0.000000  ...   
1                           2.304160                           1.261665  ...   
2                           1.622038                           0.479614  ...   
3                          19.316426                          30.501952  ...   

   TCGA-ZF-AA4X-01A-11R-A38B-07...1254  TCGA-ZF-AA51-01A-21R-A39I-07...1257  \
0                             0.000000                             0.000000   
1                             2.358371                             0.000000   
2                             0.740370                             0.257803   
3                            13.003616                            17.931997   

   TCGA-ZF-AA52-01A-12R-A39I-07...1260  TCGA-ZF-AA53-01A-11R-A39I-07...1263  \
0                             0.000000                             0.000000   
1                             0.307393                             0.000000   
2                             0.757887                             0.289427   
3                            14.704337                            21.384294   

   TCGA-ZF-AA54-01A-11R-A39I-07...1266  TCGA-ZF-AA56-01A-31R-A39I-07...1269  \
0                             0.000000                             0.145294   
1                             0.357629                             0.448098   
2                             0.452230                             0.440623   
3                            14.249617                            31.748197   

   TCGA-ZF-AA58-01A-12R-A42T-07...1272  TCGA-ZF-AA5H-01A-11R-A39I-07...1275  \
0                             0.135507                             0.000000   
1                             0.123538                             0.262174   
2                             0.110450                             0.064207   
3                            34.313044                            32.379149   

   TCGA-ZF-AA5N-01A-11R-A42T-07...1278  TCGA-ZF-AA5P-01A-11R-A39I-07...1281  
0                             0.329155                             0.000000  
1                             0.000000                             0.528753  
2                             0.000000                             0.224181  
3                            20.784490                            19.201331  

[4 rows x 428 columns]

### df.T

In [46]:
df_mrna.loc[-1]  = df_mrna.columns
df_mrna.index = df_mrna.index + 1  # shifting index
df_mrna.sort_index(inplace=True)
df_mrna_t = df_mrna.T
df_mrna_t.columns = df_mrna_t.iloc[0]
df_mrna_t = df_mrna_t.drop(df_mrna_t.index[0])
df_mrna_t.rename(columns={'gene_symbol': 'PATIENT_ID'}, inplace=True)


# df_mrna_t.index = np.arange(0, df_mrna.shape[1]-1)
# df_mrna_t.reset_index(inplace=True)
df_mrna_t.head(4)

gene_symbol                                               PATIENT_ID  \
TCGA-2F-A9KO-01A-11R-A38B-07...3    TCGA-2F-A9KO-01A-11R-A38B-07...3   
TCGA-2F-A9KP-01A-11R-A38B-07...6    TCGA-2F-A9KP-01A-11R-A38B-07...6   
TCGA-2F-A9KQ-01A-11R-A38B-07...9    TCGA-2F-A9KQ-01A-11R-A38B-07...9   
TCGA-2F-A9KR-01A-11R-A38B-07...12  TCGA-2F-A9KR-01A-11R-A38B-07...12   

gene_symbol                       100130426 100133144 100134869      10357  \
TCGA-2F-A9KO-01A-11R-A38B-07...3        0.0   1.61341  2.179003  21.731101   
TCGA-2F-A9KP-01A-11R-A38B-07...6        0.0  1.349402  0.765463  25.957999   
TCGA-2F-A9KQ-01A-11R-A38B-07...9        0.0  1.124814  0.662142  19.792975   
TCGA-2F-A9KR-01A-11R-A38B-07...12       0.0  1.231898  2.484428  26.609935   

gene_symbol                            10431 136542     155060     26823  \
TCGA-2F-A9KO-01A-11R-A38B-07...3   57.752456    0.0  12.348807  0.127691   
TCGA-2F-A9KP-01A-11R-A38B-07...6   76.897486    0.0   6.053803  0.194915   
TCGA-2F-A9KQ-01A-11R-A38B-07...9   83.883269    0.0   7.117069       0.0   
TCGA-2F-A9KR-01A-11R-A38B-07...12  40.764493    0.0  14.631614   0.53799   

gene_symbol                       280660  ...      ZXDA       ZXDB       ZXDC  \
TCGA-2F-A9KO-01A-11R-A38B-07...3     0.0  ...  3.114421  11.638891  18.871091   
TCGA-2F-A9KP-01A-11R-A38B-07...6     0.0  ...  1.186798   5.005741   23.86813   
TCGA-2F-A9KQ-01A-11R-A38B-07...9     0.0  ...   1.95251   6.049712  11.703862   
TCGA-2F-A9KR-01A-11R-A38B-07...12    0.0  ...   1.39807   7.105769  25.739295   

gene_symbol                          ZYG11A     ZYG11B         ZYX      ZZEF1  \
TCGA-2F-A9KO-01A-11R-A38B-07...3   0.131784     9.6251   181.77339  17.733666   
TCGA-2F-A9KP-01A-11R-A38B-07...6   0.359707  20.008218   137.85366  13.763714   
TCGA-2F-A9KQ-01A-11R-A38B-07...9   0.048429  20.349139  148.338828   7.326388   
TCGA-2F-A9KR-01A-11R-A38B-07...12   0.04693   9.716549  190.264225    8.10389   

gene_symbol                             ZZZ3  psiTPTE22      tAKR  
TCGA-2F-A9KO-01A-11R-A38B-07...3    13.67289   0.575769       0.0  
TCGA-2F-A9KP-01A-11R-A38B-07...6   22.037554   0.255052       0.0  
TCGA-2F-A9KQ-01A-11R-A38B-07...9    16.44073   0.214488  0.033129  
TCGA-2F-A9KR-01A-11R-A38B-07...12  23.552149  25.658375       0.0  

[4 rows x 20532 columns]

In [47]:
print(df_mrna_t['PATIENT_ID'][0][:12])
df_mrna_t['PATIENT_ID'] = df_mrna_t['PATIENT_ID'].str[:12]
df_mrna_t

TCGA-2F-A9KO


gene_symbol                            PATIENT_ID 100130426 100133144  \
TCGA-2F-A9KO-01A-11R-A38B-07...3     TCGA-2F-A9KO       0.0   1.61341   
TCGA-2F-A9KP-01A-11R-A38B-07...6     TCGA-2F-A9KP       0.0  1.349402   
TCGA-2F-A9KQ-01A-11R-A38B-07...9     TCGA-2F-A9KQ       0.0  1.124814   
TCGA-2F-A9KR-01A-11R-A38B-07...12    TCGA-2F-A9KR       0.0  1.231898   
TCGA-2F-A9KT-01A-11R-A38B-07...15    TCGA-2F-A9KT       0.0  1.145208   
...                                           ...       ...       ...   
TCGA-ZF-AA56-01A-31R-A39I-07...1269  TCGA-ZF-AA56  0.145294  0.448098   
TCGA-ZF-AA58-01A-12R-A42T-07...1272  TCGA-ZF-AA58  0.135507  0.123538   
TCGA-ZF-AA5H-01A-11R-A39I-07...1275  TCGA-ZF-AA5H       0.0  0.262174   
TCGA-ZF-AA5N-01A-11R-A42T-07...1278  TCGA-ZF-AA5N  0.329155       0.0   
TCGA-ZF-AA5P-01A-11R-A39I-07...1281  TCGA-ZF-AA5P       0.0  0.528753   

gene_symbol                         100134869      10357       10431 136542  \
TCGA-2F-A9KO-01A-11R-A38B-07...3     2.179003  21.731101   57.752456    0.0   
TCGA-2F-A9KP-01A-11R-A38B-07...6     0.765463  25.957999   76.897486    0.0   
TCGA-2F-A9KQ-01A-11R-A38B-07...9     0.662142  19.792975   83.883269    0.0   
TCGA-2F-A9KR-01A-11R-A38B-07...12    2.484428  26.609935   40.764493    0.0   
TCGA-2F-A9KT-01A-11R-A38B-07...15    1.050426  26.206662    84.14414    0.0   
...                                       ...        ...         ...    ...   
TCGA-ZF-AA56-01A-31R-A39I-07...1269  0.440623  31.748197   46.097386    0.0   
TCGA-ZF-AA58-01A-12R-A42T-07...1272   0.11045  34.313044  120.668203    0.0   
TCGA-ZF-AA5H-01A-11R-A39I-07...1275  0.064207  32.379149   88.711169    0.0   
TCGA-ZF-AA5N-01A-11R-A42T-07...1278       0.0   20.78449   68.964893    0.0   
TCGA-ZF-AA5P-01A-11R-A39I-07...1281  0.224181  19.201331   62.785841    0.0   

gene_symbol                             155060     26823 280660  ...  \
TCGA-2F-A9KO-01A-11R-A38B-07...3     12.348807  0.127691    0.0  ...   
TCGA-2F-A9KP-01A-11R-A38B-07...6      6.053803  0.194915    0.0  ...   
TCGA-2F-A9KQ-01A-11R-A38B-07...9      7.117069       0.0    0.0  ...   
TCGA-2F-A9KR-01A-11R-A38B-07...12    14.631614   0.53799    0.0  ...   
TCGA-2F-A9KT-01A-11R-A38B-07...15     7.084879       0.0    0.0  ...   
...                                        ...       ...    ...  ...   
TCGA-ZF-AA56-01A-31R-A39I-07...1269   2.550056       0.0    0.0  ...   
TCGA-ZF-AA58-01A-12R-A42T-07...1272    5.44097       0.0    0.0  ...   
TCGA-ZF-AA5H-01A-11R-A39I-07...1275   4.021965       0.0    0.0  ...   
TCGA-ZF-AA5N-01A-11R-A42T-07...1278   5.226907   0.20521    0.0  ...   
TCGA-ZF-AA5P-01A-11R-A39I-07...1281   6.289441   0.11602    0.0  ...   

gene_symbol                              ZXDA       ZXDB       ZXDC    ZYG11A  \
TCGA-2F-A9KO-01A-11R-A38B-07...3     3.114421  11.638891  18.871091  0.131784   
TCGA-2F-A9KP-01A-11R-A38B-07...6     1.186798   5.005741   23.86813  0.359707   
TCGA-2F-A9KQ-01A-11R-A38B-07...9      1.95251   6.049712  11.703862  0.048429   
TCGA-2F-A9KR-01A-11R-A38B-07...12     1.39807   7.105769  25.739295   0.04693   
TCGA-2F-A9KT-01A-11R-A38B-07...15    0.549835   4.359621  14.494392  0.140381   
...                                       ...        ...        ...       ...   
TCGA-ZF-AA56-01A-31R-A39I-07...1269  0.667266   3.239867  11.665513  0.043098   
TCGA-ZF-AA58-01A-12R-A42T-07...1272  1.043357   3.493074  12.717264  2.969045   
TCGA-ZF-AA5H-01A-11R-A39I-07...1275  0.567171   2.738361   7.399821   0.00775   
TCGA-ZF-AA5N-01A-11R-A42T-07...1278   0.30285   1.342424   7.442809  0.260585   
TCGA-ZF-AA5P-01A-11R-A39I-07...1281  0.723405   3.167222  12.438441  0.290826   

gene_symbol                             ZYG11B         ZYX      ZZEF1  \
TCGA-2F-A9KO-01A-11R-A38B-07...3        9.6251   181.77339  17.733666   
TCGA-2F-A9KP-01A-11R-A38B-07...6     20.008218   137.85366  13.763714   
TCGA-2F-A9KQ-01A-11R-A38B-07...9     20.349139  148.338828   7.326388   
TCGA-2F-A9KR-01A-11R-A38B-07...12     9.7165

### now let us check duplicates

In [57]:
nodup = df_mrna_t[~df_mrna_t["PATIENT_ID"].duplicated()]
nodup

gene_symbol                            PATIENT_ID 100130426 100133144  \
TCGA-2F-A9KO-01A-11R-A38B-07...3     TCGA-2F-A9KO       0.0   1.61341   
TCGA-2F-A9KP-01A-11R-A38B-07...6     TCGA-2F-A9KP       0.0  1.349402   
TCGA-2F-A9KQ-01A-11R-A38B-07...9     TCGA-2F-A9KQ       0.0  1.124814   
TCGA-2F-A9KR-01A-11R-A38B-07...12    TCGA-2F-A9KR       0.0  1.231898   
TCGA-2F-A9KT-01A-11R-A38B-07...15    TCGA-2F-A9KT       0.0  1.145208   
...                                           ...       ...       ...   
TCGA-ZF-AA56-01A-31R-A39I-07...1269  TCGA-ZF-AA56  0.145294  0.448098   
TCGA-ZF-AA58-01A-12R-A42T-07...1272  TCGA-ZF-AA58  0.135507  0.123538   
TCGA-ZF-AA5H-01A-11R-A39I-07...1275  TCGA-ZF-AA5H       0.0  0.262174   
TCGA-ZF-AA5N-01A-11R-A42T-07...1278  TCGA-ZF-AA5N  0.329155       0.0   
TCGA-ZF-AA5P-01A-11R-A39I-07...1281  TCGA-ZF-AA5P       0.0  0.528753   

gene_symbol                         100134869      10357       10431 136542  \
TCGA-2F-A9KO-01A-11R-A38B-07...3     2.179003  21.731101   57.752456    0.0   
TCGA-2F-A9KP-01A-11R-A38B-07...6     0.765463  25.957999   76.897486    0.0   
TCGA-2F-A9KQ-01A-11R-A38B-07...9     0.662142  19.792975   83.883269    0.0   
TCGA-2F-A9KR-01A-11R-A38B-07...12    2.484428  26.609935   40.764493    0.0   
TCGA-2F-A9KT-01A-11R-A38B-07...15    1.050426  26.206662    84.14414    0.0   
...                                       ...        ...         ...    ...   
TCGA-ZF-AA56-01A-31R-A39I-07...1269  0.440623  31.748197   46.097386    0.0   
TCGA-ZF-AA58-01A-12R-A42T-07...1272   0.11045  34.313044  120.668203    0.0   
TCGA-ZF-AA5H-01A-11R-A39I-07...1275  0.064207  32.379149   88.711169    0.0   
TCGA-ZF-AA5N-01A-11R-A42T-07...1278       0.0   20.78449   68.964893    0.0   
TCGA-ZF-AA5P-01A-11R-A39I-07...1281  0.224181  19.201331   62.785841    0.0   

gene_symbol                             155060     26823 280660  ...  \
TCGA-2F-A9KO-01A-11R-A38B-07...3     12.348807  0.127691    0.0  ...   
TCGA-2F-A9KP-01A-11R-A38B-07...6      6.053803  0.194915    0.0  ...   
TCGA-2F-A9KQ-01A-11R-A38B-07...9      7.117069       0.0    0.0  ...   
TCGA-2F-A9KR-01A-11R-A38B-07...12    14.631614   0.53799    0.0  ...   
TCGA-2F-A9KT-01A-11R-A38B-07...15     7.084879       0.0    0.0  ...   
...                                        ...       ...    ...  ...   
TCGA-ZF-AA56-01A-31R-A39I-07...1269   2.550056       0.0    0.0  ...   
TCGA-ZF-AA58-01A-12R-A42T-07...1272    5.44097       0.0    0.0  ...   
TCGA-ZF-AA5H-01A-11R-A39I-07...1275   4.021965       0.0    0.0  ...   
TCGA-ZF-AA5N-01A-11R-A42T-07...1278   5.226907   0.20521    0.0  ...   
TCGA-ZF-AA5P-01A-11R-A39I-07...1281   6.289441   0.11602    0.0  ...   

gene_symbol                              ZXDA       ZXDB       ZXDC    ZYG11A  \
TCGA-2F-A9KO-01A-11R-A38B-07...3     3.114421  11.638891  18.871091  0.131784   
TCGA-2F-A9KP-01A-11R-A38B-07...6     1.186798   5.005741   23.86813  0.359707   
TCGA-2F-A9KQ-01A-11R-A38B-07...9      1.95251   6.049712  11.703862  0.048429   
TCGA-2F-A9KR-01A-11R-A38B-07...12     1.39807   7.105769  25.739295   0.04693   
TCGA-2F-A9KT-01A-11R-A38B-07...15    0.549835   4.359621  14.494392  0.140381   
...                                       ...        ...        ...       ...   
TCGA-ZF-AA56-01A-31R-A39I-07...1269  0.667266   3.239867  11.665513  0.043098   
TCGA-ZF-AA58-01A-12R-A42T-07...1272  1.043357   3.493074  12.717264  2.969045   
TCGA-ZF-AA5H-01A-11R-A39I-07...1275  0.567171   2.738361   7.399821   0.00775   
TCGA-ZF-AA5N-01A-11R-A42T-07...1278   0.30285   1.342424   7.442809  0.260585   
TCGA-ZF-AA5P-01A-11R-A39I-07...1281  0.723405   3.167222  12.438441  0.290826   

gene_symbol                             ZYG11B         ZYX      ZZEF1  \
TCGA-2F-A9KO-01A-11R-A38B-07...3        9.6251   181.77339  17.733666   
TCGA-2F-A9KP-01A-11R-A38B-07...6     20.008218   137.85366  13.763714   
TCGA-2F-A9KQ-01A-11R-A38B-07...9     20.349139  148.338828   7.326388   
TCGA-2F-A9KR-01A-11R-A38B-07...12     9.7165

In [63]:
df_mrna_t['is_dup'] = df_mrna_t.duplicated(subset=['PATIENT_ID'], keep=False)

In [64]:
df_mrna_t.loc[df_mrna_t['is_dup'] == True]


gene_symbol                           PATIENT_ID 100130426 100133144  \
TCGA-BL-A13J-11A-13R-A10U-07...81   TCGA-BL-A13J       0.0  0.210345   
TCGA-BL-A13J-01A-11R-A10U-07...84   TCGA-BL-A13J       0.0   1.70432   
TCGA-BT-A20N-11A-11R-A14Y-07...102  TCGA-BT-A20N       0.0  0.301691   
TCGA-BT-A20N-01A-11R-A14Y-07...105  TCGA-BT-A20N       0.0    0.5557   
TCGA-BT-A20Q-11A-11R-A14Y-07...114  TCGA-BT-A20Q       0.0  0.417331   
TCGA-BT-A20Q-01A-11R-A14Y-07...117  TCGA-BT-A20Q       0.0  1.114282   
TCGA-BT-A20R-11A-11R-A16R-07...120  TCGA-BT-A20R       0.0  1.166343   
TCGA-BT-A20R-01A-12R-A16R-07...123  TCGA-BT-A20R       0.0  1.146664   
TCGA-BT-A20U-11A-11R-A14Y-07...129  TCGA-BT-A20U       0.0  1.769722   
TCGA-BT-A20U-01A-11R-A14Y-07...132  TCGA-BT-A20U       0.0  1.524552   
TCGA-BT-A20W-11A-11R-A14Y-07...138  TCGA-BT-A20W       0.0   0.45949   
TCGA-BT-A20W-01A-21R-A14Y-07...141  TCGA-BT-A20W       0.0  0.563223   
TCGA-BT-A2LA-11A-11R-A18C-07...147  TCGA-BT-A2LA       0.0  0.642488   
TCGA-BT-A2LA-01A-11R-A18C-07...150  TCGA-BT-A2LA       0.0    1.8965   
TCGA-BT-A2LB-11A-11R-A18C-07...153  TCGA-BT-A2LB       0.0  1.921218   
TCGA-BT-A2LB-01A-11R-A18C-07...156  TCGA-BT-A2LB       0.0       0.0   
TCGA-CU-A0YN-11A-11R-A10U-07...261  TCGA-CU-A0YN       0.0  0.347162   
TCGA-CU-A0YN-01A-21R-A10U-07...264  TCGA-CU-A0YN  0.149225  1.704357   
TCGA-CU-A0YR-11A-13R-A10U-07...270  TCGA-CU-A0YR       0.0  0.696596   
TCGA-CU-A0YR-01A-12R-A10U-07...273  TCGA-CU-A0YR       0.0  0.767832   
TCGA-GC-A3BM-11A-11R-A22U-07...729  TCGA-GC-A3BM       0.0  1.641141   
TCGA-GC-A3BM-01A-11R-A22U-07...732  TCGA-GC-A3BM       0.0  1.870221   
TCGA-GC-A3WC-11A-11R-A22U-07...750  TCGA-GC-A3WC       0.0  0.253008   
TCGA-GC-A3WC-01A-31R-A22U-07...753  TCGA-GC-A3WC       0.0  4.968682   
TCGA-GC-A6I3-11A-11R-A31N-07...765  TCGA-GC-A6I3       0.0       0.0   
TCGA-GC-A6I3-01A-11R-A31N-07...768  TCGA-GC-A6I3       0.0  0.321573   
TCGA-GD-A2C5-11A-11R-A180-07...771  TCGA-GD-A2C5       0.0  0.468907   
TCGA-GD-A2C5-01A-12R-A180-07...774  TCGA-GD-A2C5       0.0  2.031788   
TCGA-GD-A3OP-11A-11R-A220-07...777  TCGA-GD-A3OP       0.0  1.358474   
TCGA-GD-A3OP-01A-21R-A220-07...780  TCGA-GD-A3OP       0.0  1.928463   
TCGA-GD-A3OQ-11A-21R-A220-07...783  TCGA-GD-A3OQ       0.0  3.897811   
TCGA-GD-A3OQ-01A-32R-A220-07...786  TCGA-GD-A3OQ       0.0  0.497523   
TCGA-K4-A3WV-11A-21R-A22U-07...891  TCGA-K4-A3WV       0.0  0.644349   
TCGA-K4-A3WV-01A-11R-A22U-07...894  TCGA-K4-A3WV       0.0  5.664856   
TCGA-K4-A54R-11A-11R-A26T-07...903  TCGA-K4-A54R       0.0  1.095031   
TCGA-K4-A54R-01A-11R-A26T-07...906  TCGA-K4-A54R       0.0  0.338861   
TCGA-K4-A5RI-11A-11R-A28M-07...912  TCGA-K4-A5RI       0.0  0.102158   
TCGA-K4-A5RI-01A-11R-A28M-07...915  TCGA-K4-A5RI       0.0  0.833669   

gene_symbol                        100134869      10357       10431 136542  \
TCGA-BL-A13J-11A-13R-A10U-07...81    0.50795  22.308088   53.257721    0.0   
TCGA-BL-A13J-01A-11R-A10U-07...84   0.860396  27.338578  112.325948    0.0   
TCGA-BT-A20N-11A-11R-A14Y-07...102  0.321618  18.677608   59.747054    0.0   
TCGA-BT-A20N-01A-11R-A14Y-07...105  0.135634  25.332553  139.973861    0.0   
TCGA-BT-A20Q-11A-11R-A14Y-07...114  0.577184  26.375325   54.174538    0.0   
TCGA-BT-A20Q-01A-11R-A14Y-07...117  0.938832  35.497915   86.758907    0.0   
TCGA-BT-A20R-11A-11R-A16R-07...120  0.396708  15.992367   58.095019    0.0   
TCGA-BT-A20R-01A-12R-A16R-07...123   0.65744     9.8813   85.686509    0.0   
TCGA-BT-A20U-11A-11R-A14Y-07...129  0.710135  19.250042   60.739847    0.0   
TCGA-BT-A20U-01A-11R-A14Y-07...132  1.198071  31.444971   63.307949    0.0   
TCGA-BT-A20W-11A-11R-A14Y-07...138  0.086283  26.207118   94.832968    0.0   
TCGA-BT-A20W-01A-21R-A14Y-07...141  0.351726   34.30548  138.227126    0.0   
TCGA-BT-A2LA-11A-11R-A18C-07...147  0.140364  15.487324   68.460939    0.0   
TCGA-BT-A2LA-01A-11R-A18C-07...150   1.88269  24.984511    56.33266    0.0   
TCGA-BT-A2LB-11A-11R-

In [67]:
### Merge

In [9]:
df = pd.merge(df_cp, df_mrna_t, on="#Patient Identifier", how="inner")

In [10]:
print(df.shape)
df.head(5)

(407, 20568)


#Patient Identifier Subtype TCGA PanCanAtlas Cancer Type Acronym  \
0        TCGA-2F-A9KO    BLCA                                 BLCA   
1        TCGA-2F-A9KP    BLCA                                 BLCA   
2        TCGA-2F-A9KQ    BLCA                                 BLCA   
3        TCGA-2F-A9KR    BLCA                                 BLCA   
4        TCGA-2F-A9KT    BLCA                                 BLCA   

                       Other Patient ID Diagnosis Age     Sex  \
0  DD3A6357-9087-44B2-9956-F981E0DE6F1C            63    Male   
1  65D1EAEC-28DB-4A41-A4DB-1A710FCB24AD            66    Male   
2  25EAF3F9-C364-423C-AAAE-925E7B393AFC            69    Male   
3  F6D916B0-8E4C-49CB-A0D4-883908F3284F            59  Female   
4  BC6C516B-591E-4950-B6B0-DECAFA666F4E            83    Male   

  Neoplasm Disease Stage American Joint Committee on Cancer Code  \
0                                           STAGE IV               
1                                           STAGE IV               
2                                          STAGE III               
3                                          STAGE III               
4                                           STAGE II               

  American Joint Committee on Cancer Publication Version Type  \
0                                                6TH            
1                                                7TH            
2                                                6TH            
3                                                5TH            
4                                                6TH            

  Last Communication Contact from Initial Pathologic Diagnosis Date  \
0                                                678                  
1                                                NaN                  
2                                               2886                  
3                                               3148                  
4                                               3314                  

  Birth from Initial Pathologic Diagnosis Date  ...   ZWILCH    ZWINT  \
0                                       -23323  ...  694.857  554.519   
1                                       -24428  ...  718.119  969.621   
2                                       -25259  ...  532.536  795.086   
3                                       -21848  ...  616.848  568.412   
4                                       -30520  ...  726.625  1087.05   

      ZXDA     ZXDB     ZXDC  ZYG11A   ZYG11B      ZYX    ZZEF1     ZZZ3  
0  102.624  776.676  1450.73  5.8309  697.959  4262.39  1787.76  572.595  
1   37.037  316.688  1420.72  17.062  1526.43  3105.29  1467.75  860.591  
2  60.2689  378.303  873.435  1.8544  1571.16  3275.38   750.58  621.233  
3  45.5869  470.115  1924.15  1.8995   748.86  4465.62  910.788  935.481  
4   17.524  281.515  938.383  6.7835  878.462  3449.41  850.198  630.865  

[5 rows x 20568 columns]

In [11]:
print(len(np.unique(np.array((df['#Patient Identifier']))) ))

407


Now We know that there is no duplicated Patient IDs as there are only 407 unique number.
now let us save the data into a CSV

In [ ]:
df.to_csv('data/data.csv')